### PR TITLE
feat(amazonq): loosen inline suggestion restriction for json/yaml files

### PR DESCRIPTION
**Pull request by Will-ShaoHua**
VSC PR https://github.com/aws/aws-toolkit-vscode/pull/5733

YAML files

- Provide completions for all YAML files.

- Remove any current CloudFormation-only restrictions for YAML

JSON files

- Provide completions for a JSON file if the filename matches one of the below names or the file content matches the CloudFormation regexp.
```
    app.json
    appsettings.json
    bower.json
    composer.json
    db.json
    manifest.json
    package.json
    schema.json
    settings.json
    tsconfig.json
    vcpkg.json
```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist
- [ ] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [ ] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/editor/CodeWhispererEditorUtil.kt [hunk 1]
```diff
@@ -106,16 +109,15 @@ object CodeWhispererEditorUtil {
...
      */
-    fun checkLeftContextKeywordsForJsonAndYaml(leftContext: String, language: String): Boolean = (
-        (language == CodeWhispererJson.INSTANCE.languageId) ||
-            (language == CodeWhispererYaml.INSTANCE.languageId)
-        ) &&
-        (
-            (!CodeWhispererConstants.AWSTemplateKeyWordsRegex.containsMatchIn(leftContext)) &&
-                (!CodeWhispererConstants.AWSTemplateCaseInsensitiveKeyWordsRegex.containsMatchIn(leftContext.lowercase(Locale.getDefault())))
-            )
+    fun isConfigFileIfJsonFile(fileName: String, leftContext: String, language: CodeWhispererProgrammingLanguage): Boolean =
```
Comment (andrewyuq) [inline plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/editor/CodeWhispererEditorUtil.kt:114]: `isConfigFileIfJsonFile` method name a bit confusing

Comment (Will-ShaoHua) [reply plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/editor/CodeWhispererEditorUtil.kt:114]: makes sense, will rename it to `isJsonFormatSupported` maybe?

Comment (andrewyuq) [reply plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/editor/CodeWhispererEditorUtil.kt:114]: isJsonFileSupported

Comment (Will-ShaoHua) [reply plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/editor/CodeWhispererEditorUtil.kt]: i did this instead `isSupportedJsonFormat`

**Comment by Will-ShaoHua** [comment]
updated case ignore

File: plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererService.kt [hunk 2]
```diff
@@ -171,7 +171,7 @@ class CodeWhispererService(private val cs: CoroutineScope) : Disposable {
 
         val language = requestContext.fileContextInfo.programmingLanguage
         val leftContext = requestContext.fileContextInfo.caretContext.leftFileContext
-        if (!language.isCodeCompletionSupported() || (checkLeftContextKeywordsForJsonAndYaml(leftContext, language.languageId))) {
+        if (!language.isCodeCompletionSupported() || !isSupportedJsonFormat(requestContext.fileContextInfo.filename, leftContext, language)) {
```
Comment (zixlin7) [inline plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererService.kt:174]: I might be missing something, where did we add yaml support?

**Review by andrewyuq — APPROVED** [review]

File: plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererService.kt (hunk 2 above)
Comment (Will-ShaoHua) [reply plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererService.kt]: but with some condition, so this PR simply remove the restriction for YAML

**Review by zixlin7 — APPROVED** [review]

File: plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/editor/CodeWhispererEditorUtil.kt [hunk 3]
```diff
@@ -106,16 +109,14 @@ object CodeWhispererEditorUtil {
...
-    fun checkLeftContextKeywordsForJsonAndYaml(leftContext: String, language: String): Boolean = (
-        (language == CodeWhispererJson.INSTANCE.languageId) ||
-            (language == CodeWhispererYaml.INSTANCE.languageId)
-        ) &&
-        (
-            (!CodeWhispererConstants.AWSTemplateKeyWordsRegex.containsMatchIn(leftContext)) &&
-                (!CodeWhispererConstants.AWSTemplateCaseInsensitiveKeyWordsRegex.containsMatchIn(leftContext.lowercase(Locale.getDefault())))
-            )
+    fun isSupportedJsonFormat(fileName: String, leftContext: String, language: CodeWhispererProgrammingLanguage): Boolean {
+        assert(language is CodeWhispererJson)
```
[the remark this answers is not in the document]
Comment (Will-ShaoHua) [reply plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/editor/CodeWhispererEditorUtil.kt:115]: ack

**Review by manodnyab — APPROVED** [review]

**Pull request merged**